src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -221,7 +222,7 @@ private void ComputeParameters()
                 _declarationDependencies.AddAll(diagnostics.DependenciesBag);
                 diagnostics.Free();
                 _lazyIsVarArg = isVararg;
-                _lazyParameters = parameters;
+                RoslynImmutableInterlocked.VolatileWrite(ref _lazyParameters, parameters);
```

Comment:
Nothing prevents writes _within_ a lock from being observed in alternate orders by other threads, when those other threads don't use a `lock` to check themselves. Since `_lazyParameters` is checked for `Default` outside of a lock as the sentinel, we therefore need to make sure that it's the last write inside this `lock` or we could still race.